homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,221 @@
+                    "rtsp_flags": "prefer_tcp",
+                    "stimeout": "5000000",
+                }
+            if rtsp_transport := info.get(CONF_RTSP_TRANSPORT):
+                stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = rtsp_transport
+            _LOGGER.debug("Attempting to open stream %s", stream_source)
+            container = await self.hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+            return True, ""
+        # pylint: disable=c-extension-no-member
+        except av.error.FileNotFoundError:
+            return False, "stream_no_route_to_host"
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return False, "stream_unauthorised"
+        except (KeyError, IndexError):
+            return False, "stream_novideo"
+        except PermissionError:
+            return False, "stream_not_permitted"
+        except OSError as err:
+            if "No route to host" in str(err):
+                return False, "stream_no_route_to_host"
+            if "Input/output error" in str(err):
+                return False, "stream_io_error"
+            raise err
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): str,
+            vol.Optional(
+                CONF_STILL_IMAGE_URL,
+                description={
+                    "suggested_value": user_input.get(CONF_STILL_IMAGE_URL, "")
+                },
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                description={"suggested_value": user_input.get(CONF_STREAM_SOURCE, "")},
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT)
+            ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input.get(CONF_AUTHENTICATION)
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                description={"suggested_value": user_input.get(CONF_USERNAME, "")},
+            ): str,
+            vol.Optional(
+                CONF_PASSWORD,
+                description={"suggested_value": user_input.get(CONF_PASSWORD, "")},
+            ): str,
+            vol.Optional(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE),
+            ): bool,
+            vol.Optional(
+                CONF_CONTENT_TYPE, default=user_input.get(CONF_CONTENT_TYPE)
+            ): str,
```

Comment:
Is this required to setup the camera?

Reply:
Yes, it defines the image type that is returned when the browser tries to view the image after setup.  But digging deeper it can be derived automatically - so I don't think it is necessary on the config flow.  Will remove it.

Reply:
Done.